mypy/plugins/dataclasses.py

```diff
@@ -380,6 +382,26 @@ def reset_init_only_vars(self, info: TypeInfo, attributes: list[DataclassAttribu
                             # recreate a symbol node for this attribute.
                             lvalue.node = None
 
+    @staticmethod
```

Comment:
Any reason to make this a `@staticmethod`? I think it only makes sense as an instance method. I would recommend to remove this decorator.